crates/bdk/src/wallet/mod.rs

```diff
@@ -2353,6 +2386,70 @@ impl<D> Wallet<D> {
     pub fn local_chain(&self) -> &LocalChain {
         &self.chain
     }
+
+    /// Create a [`SyncRequest`] for this wallet.
+    ///
+    /// This is the first step when performing a spk-based wallet sync, the returned [`SyncRequest`] collects
+    /// the wallet keychain all or unused script pub keys, unconfirmed transaction id, UTXOs and local
+    /// chain checkpoint data needed to start a blockchain sync with a blockchain client. For a faster
+    /// sync set `unused_spks_only` to true to only get updates for unused wallet script pub keys (addresses).
+    pub fn sync_request(&self, unused_spks_only: bool) -> SyncRequest {
+        let checkpoint = self.latest_checkpoint();
+
+        // Sync only unused SPKs
+        let spks = if unused_spks_only {
+            self.spk_index()
+                .unused_spks()
+                .map(|(_keychain, _index, script)| ScriptBuf::from(script))
+                .collect::<Vec<ScriptBuf>>()
+        }
+        // Sync all SPKs
+        else {
+            self.spk_index()
+                .revealed_spks()
+                .map(|(_keychain, _index, script)| ScriptBuf::from(script))
+                .collect::<Vec<ScriptBuf>>()
+        };
+
+        // Sync UTXOs
+        // We want to search for whether our UTXOs are spent, and spent by which transaction.
+        let outpoints: Vec<OutPoint> = self.list_unspent().map(|utxo| utxo.outpoint).collect();
+
+        // Sync unconfirmed TX
+        // We want to search for whether our unconfirmed transactions are now confirmed.
+        let txids: Vec<Txid> = self
+            .transactions()
+            .filter(|canonical_tx| !canonical_tx.chain_position.is_confirmed())
+            .map(|canonical_tx| canonical_tx.tx_node.txid)
+            .collect();
```

Comment:
Performance note: In theory there are faster ways to get the set of unconfirmed transactions in the graph. We haven't implemented them though.